src/components/pod/pod.component.tsx

```diff
@@ -0,0 +1,267 @@
+  /** Determines if the editable pod content should be full width */
+  editContentFullWidth?: boolean;
+  /** Determines if the edit button should be hidden until the user hovers over the content */
+  displayEditButtonOnHover?: boolean;
+  /** Determines if clicking the pod content calls the onEdit action */
+  triggerEditOnContent?: boolean;
+  /** Resets edit button styles to an older version */
+  internalEditButton?: boolean;
+  /** Sets Pod height, number is changed to pixels and string is passed as raw css value */
+  height?: string | number;
+}
+
+const Pod = React.forwardRef<HTMLDivElement, PodProps>(
+  (
+    {
+      alignTitle = "left",
+      border = true,
+      children,
+      className,
+      displayEditButtonOnHover,
+      editContentFullWidth,
+      footer,
+      height,
+      internalEditButton,
+      onDelete,
+      onEdit,
+      onUndo,
+      size = "medium",
+      softDelete,
+      subtitle,
+      title,
+      triggerEditOnContent,
+      variant = "primary",
+      ...rest
+    }: PodProps,
+    ref
+  ) => {
+    const [isEditFocused, setEditFocused] = useState(false);
+    const [isEditHovered, setEditHovered] = useState(false);
+    const [isDeleteFocused, setDeleteFocused] = useState(false);
+    const [isDeleteHovered, setDeleteHovered] = useState(false);
+    const [isUndoFocused, setUndoFocused] = useState(false);
+    const [isUndoHovered, setUndoHovered] = useState(false);
+
+    const l = useLocale();
+    const shouldContentHaveEditEvents = useMemo(
+      () => !!(onEdit && (triggerEditOnContent || displayEditButtonOnHover)),
+      [displayEditButtonOnHover, onEdit, triggerEditOnContent]
+    );
+
+    const processPodAction = (
+      action: React.EventHandler<
+        React.KeyboardEvent<HTMLElement> | React.MouseEvent<HTMLElement>
+      >
+    ) => (
+      ev: React.KeyboardEvent<HTMLElement> | React.MouseEvent<HTMLElement>
+    ) => {
+      if (
+        Event.isKeyboardEvent(ev) &&
+        Event.isEventType(ev, "keydown") &&
+        !Event.isEnterKey(ev)
+      )
+        return;
+
+      ev.preventDefault();
+      action(ev);
+    };
+
+    const editEvents = {
+      onMouseEnter: () => setEditHovered(true),
+      onMouseLeave: () => setEditHovered(false),
+      onFocus: () => setEditFocused(true),
+      onBlur: () => setEditFocused(false),
+      ...(typeof onEdit === "function" && {
+        onClick: processPodAction(onEdit),
+        onKeyDown: processPodAction(onEdit),
+      }),
+    };
+
+    return (
+      <StyledPod
+        alignTitle={alignTitle}
+        className={className}
+        internalEditButton={internalEditButton}
+        height={typeof height === "number" ? `${height}px` : height}
+        ref={ref}
+        {...tagComponent("pod", rest)}
+        {...rest}
+      >
+        <StyledBlock
+          data-element="block"
+          contentTriggersEdit={shouldContentHaveEditEvents}
+          hasButtons={!!(onEdit || onDelete || onUndo)}
```

Comment:
Might be worth assigning this boolean to a variable as it's used below as well (line 199)

eg: const hasButtons = !!(onEdit || onDelete || onUndo);